app/src/main/java/vn/realtest/stock/justanewproject/Adapter/HoseAdapter.java

```diff
@@ -0,0 +1,98 @@
+package vn.realtest.stock.justanewproject.Adapter;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.content.LocalBroadcastManager;
+import android.support.v7.widget.CardView;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.List;
+
+import vn.realtest.stock.justanewproject.Data.MarketStock;
+import vn.realtest.stock.justanewproject.R;
+
+/**
+ * Created by Paul on 3/13/2018.
+ */
+
+public class HoseAdapter extends RecyclerView.Adapter<HoseAdapter.MarketStockViewHolder> {
```

Comment:
all these kinds of adapter can be extended from a general adapter like `ListAdapter` because it handles all thing in the same but the `id_san`. Please consider refactoring if you have time